sympy/solvers/solveset.py

```diff
@@ -932,7 +932,7 @@ def solveset(f, symbol=None, domain=S.Complexes):
             raise NotImplementedError(filldedent('''
                 relationship between value and 0 is unknown: %s''' % b))
 
-    if f.has(Max) or f.has(Min):
+    if not f.has(Abs):
```

Comment:
I think there are (at least) two more robust ways to do this:

1) mask off the Abs with dummies, rewrite, then re-substitute in the Abs
2) extract and rewrite only functions that have x in their free symbols

What do your current mods give for `Max(abs(x-3)-1,x+2)-3)`? The answer should be `[-1, 1]`.

Reply:
> give for 
I mean `solveset(..., x, S.Reals)`

Reply:
>What do your current mods give for Max(abs(x-3)-1,x+2)-3)? The answer should be [-1, 1].

It gives a `ConditionSet`.(same as the current master)
..And I am working on the two ways.

Reply:
I think the most robust way to go about it is to get the `[a for a in expr.atoms(Function) if x in a.free_symbols]` and rewrite those as Piecewise and those that changed get put into a replacement dictionary that is used to replace their occurrence in the original expression and then fold that expression and solve. Doing this will keep non-x dependent functions from being rewritten. e.g.

```
>>> solve(piecewise_fold(Abs(x-1).rewrite(Piecewise)-abs(y)),x)
[Abs(y) + 1, Piecewise((-Abs(y) + 1, -Abs(y) < 0), (nan, True))]
>>> solve(piecewise_fold((Abs(x-1)-abs(y)).rewrite(Piecewise)),x)
[Piecewise((-y + 1, y > 0), (nan, True)), Piecewise((-y + 1, y < 0), (nan, True)), 
Piecewise((y + 1, y >= 0), (nan, True)), Piecewise((y + 1, y < 0), (nan, True))]
```

Reply:
@smichr The above modification is just resolves problem with  the multivariate equations. I think the idea of masking off(as mentioned above) should also be implemented in addition to it. Else the `test_issue_9565` would fail as it fails when the `Abs` is converted to `Piecewise`.